rc/tools/git.kak

```diff
@@ -106,22 +112,31 @@ define-command -params 1.. \
             update-diff \
         ;
     else
+        # Without maintaining the state we cannot know the work tree.
+        # Autocompletion will work only for the default case when git work tree is at `cwd`
+        # Piping to /dev/null to silent noisy git errors
         case "$1" in
-            commit) printf -- "--amend\n--no-edit\n--all\n--reset-author\n--fixup\n--squash\n"; git ls-files -m ;;
-            add) git ls-files -dmo --exclude-standard ;;
+            commit) printf -- "--amend\n--no-edit\n--all\n--reset-author\n--fixup\n--squash\n"; git ls-files -m 2>/dev/null ;;
+            add) git ls-files -dmo --exclude-standard 2>/dev/null ;;
             apply) printf -- "--reverse\n--cached\n--index\n" ;;
-            grep|edit) git ls-files -c --recurse-submodules ;;
+            grep|edit) git ls-files -c --recurse-submodules 2>/dev/null ;;
         esac
     fi
   } \
-  git %{ evaluate-commands %sh{
-    cd_bufdir() {
-        dirname_buffer="${kak_buffile%/*}"
-        cd "${dirname_buffer}" 2>/dev/null || {
-            printf 'fail Unable to change the current working directory to: %s\n' "${dirname_buffer}"
-            exit 1
-        }
-    }
+  git %{
+  set-option buffer git_work_tree %sh{
```

Comment:
I don't think we should set the option on the buffer where the git command is invoked, unless there is a motivation I have missed?